### PR TITLE
GITHUB-1011 - UnnecessaryretryFailed running within a loop of stored instances. Take out the loop.

### PR DESCRIPTION
**Pull request by matzinger0670**
Unnecessary retryFailed running within a loop of stored instances. Take out the loop. (issue #1011)

### CODE REVIEW

**Comment by juherr** [comment]
Sadly, the pull-request is not clean enough and it is not possible to review it online. 

Could you rework it in order to have the diff as small as possible ? 
Could you add a test case too ?

**Comment by matzinger0670** [comment]
I have made changes, but could you please give me a hint how I can run testCase in Eclipse with JUnit?

**Comment by matzinger0670** [comment]
I have reworked my pull-request.

File: src/test/java/test/retryAnalyzer/InvocationCountTest.java [hunk 1]
```diff
@@ -65,6 +70,39 @@ public void checkInvocationCounts() {
     assertEquals(invocations.count("failAfterThreeRetries"), 4);
   }
 
+  @Test (
+			retryAnalyzer = InvocationCountTest.class,
+			dataProvider = "dataProvider3")
+  public void retryWithDataProvider(String tc, int paf, boolean test) {
```
Comment (juherr) [inline src/test/java/test/retryAnalyzer/InvocationCountTest.java]: As `int paf, boolean test` params are not used, maybe you can remove them?

**Comment by juherr** [comment]
I'm trying to understand why `instances` was used and why your change is working.
`instances` comes from the integration of retry-analyzer: https://github.com/cbeust/testng/commit/2347cf68a7c03a9e19054b1ebd59f31ecbc09335
In fact, I don't feel good to change something which lives from the beginning.

@cbeust Could you have a look on the issue and fix?
@VladRassokhin I saw you changed things arround this retry-analyzer code 2 years ago, maybe you can have an idea or feedback to share :)

**Comment by matzinger0670** [comment]
Any news on this issue?

**Comment by juherr** [comment]
I'm not feeling good the removal of `instances`. And I'd like @cbeust's opinion.
Could you check if a retry test exists with a `@Factory`, and add it if not?

**Comment by cbeust** [comment]
I'm not sure what the original problem is, can you elaborate?

**Comment by juherr** [comment]
@cbeust Check #1011

**Comment by cbeust** [comment]
Got it. Ok I understand that this PR fixes the problem described in #1011. Like @juherr, I'm a bit wondering if this fix can impact other areas but at this point, I'm fine merging this in.

@juherr Are you strongly opposed to this PR or just a bit concerned like I am? If the latter, I'll merge and we'll see what happens.

**Comment by juherr** [comment]
I've thought I already replied, sorry.

@cbeust Ok for me.

File: src/test/java/test/retryAnalyzer/InvocationCountTest.java [hunk 2]
```diff
@@ -65,6 +70,39 @@ public void checkInvocationCounts() {
     assertEquals(invocations.count("failAfterThreeRetries"), 4);
   }
 
+  @Test (
```
Comment (cbeust) [inline src/test/java/test/retryAnalyzer/InvocationCountTest.java]: The formatting is a bit funky here.

File: src/test/java/test/retryAnalyzer/InvocationCountTest.java [hunk 3]
```diff
@@ -126,4 +170,13 @@ public boolean retry(ITestResult result) {
       return remainingRetries.getAndDecrement() > 0;
     }
   }
+  
+  public static class RetryCountTest implements IRetryAnalyzer {
+	    private final AtomicInteger remainingRetries = new AtomicInteger(12);
```
Comment (cbeust) [inline src/test/java/test/retryAnalyzer/InvocationCountTest.java]: Same: two space indents.

File: src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java [hunk 4]
```diff
@@ -16,6 +16,7 @@
 public class RetryAnalyzerTest extends SimpleBaseTest {
     @Test
     public void testInvocationCounts() {
+
```
Comment (cbeust) [inline src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java]: Only newlines added to this file, remove it from the PR.

**Comment by matzinger0670** [comment]
...one question! because it is the first time colaborate in github. 
Are these changes merged already or have forgotten to do for that?

**Comment by juherr** [comment]
Hi, the pull request is not yet merged because we are waiting for the changes asked by @cbeust

**Pull request merged**